cit3.0-web/src/components/Page/OpportunityDeletePage/OpportunityDeletePage.js

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import { Alert } from "shared-components/build/components/alert/Alert";
+import { MdError } from "react-icons/md";
+import { useDispatch, useSelector } from "react-redux";
+import { useHistory, useLocation } from "react-router-dom";
+import { Container } from "react-bootstrap";
+import OpportunityFactory from "../../../store/factory/OpportunityFactory";
+import {
+  deleteOpportunity,
+  getOpportunity,
+  resetOpportunity,
+  setOpportunity,
+} from "../../../store/actions/opportunity";
+import ButtonRow from "../../ButtonRow/ButtonRow";
+import TextInput from "../../FormComponents/TextInput";
+
+const OpportunityDeletePage = ({ id }) => {
+  const history = useHistory();
+  const location = useLocation();
+  const dispatch = useDispatch();
+  const [confirmName, setConfirmName] = useState();
+  const opportunity = useSelector((state) => state.opportunity);
+
+  useEffect(() => {
+    let opId = id;
+    console.log(opId);
+    if (!id) {
+      const found = location.pathname.match(/(\d+)+\/?$/);
+      opId = found && parseInt(found[0], 10);
+      console.log(opId);
+    }
+    if (opId !== opportunity.id) {
+      getOpportunity(opId).then((response) => {
+        const opp = OpportunityFactory.createStateFromResponse(response.data);
+        dispatch(setOpportunity(opp));
+      });
+    }
+  }, []);
+
+  const handleDeleteOpportunity = (e) => {
+    deleteOpportunity(opportunity).then(() => {
+      dispatch(resetOpportunity());
+      dispatch(resetOpportunity());
```

Comment:
Is this function supposed to be called twice here?

Reply:
Fixed